framework/include/physics/PhysicsBase.h

```diff
@@ -0,0 +1,248 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Action.h"
+
+// We include these headers for all the derived classes that will be building objects
+#include "FEProblemBase.h"
+#include "Factory.h"
+
+#define registerPhysicsBaseTasks(app_name, derived_name)                                           \
+  registerMooseAction(app_name, derived_name, "init_physics")
+
+/**
+ * Base class to help creates an entire physics
+ */
+class PhysicsBase : public Action
+{
+public:
+  static InputParameters validParams();
+
+  PhysicsBase(const InputParameters & parameters);
+
+  /// Forwards from the action tasks to the implemented addXYZ() in the derived classes
+  /// If you need more than these:
+  /// - register your action to the new task using
+  ///   registerMooseAction("AppName", ActionClass, "task_name");
+  /// - override act and add your additional work there
```

Comment:
```suggestion
```

Reply:
what's the plan if we need a derived Physics to use additional tasks? For example if we are dealing with an app that has systems that dont exist in moose/framework
I m thinking the 'final' will have to go (either one day or now)

Reply:
While it is nice to pre-list the tasks, and safer because there's less chances for a mismatch (ignored tasks for which we implemented work is the main risk), we've never had this for actions. I could add some sort of boolean that checks that the main "act" of PhysicsBase always gets called. But that's unconventional?

Reply:
actually I have a solution for this

Reply:
added